### ISSUE TITLE
[case-mgmt-service]: failed to output all searched tickets with calling CaseManagementV1.get_cases()

### ISSUE THREAD

**Issue by DanielWu0626** [IBM/platform-services-python-sdk#77]
**Describe the bug**
as above title

**To Reproduce**
here is my python api code , I want to list all "raid alert" ticket under one customer account, actually it found 314 tickets by searching the keyword "Raid Alert", but unluckily output table only included 100 records,  Would someone take a look if this is bug or use error ? thanks !

---------------
![image](https://user-images.githubusercontent.com/35214998/105953955-ff602880-60ae-11eb-8e91-e62171182d55.png)

---------------

danws-MBP:api_samples danw$ ./getCaseList.py
100

Total <314> raid alert tickets were found

+-----------+-----------------------------------+-------------+-------------+---------------+
|  case_id  |             case_title            | case_status | hareware_id | hardware_name |
+-----------+-----------------------------------+-------------+-------------+---------------+
| CS1967301 | Required Maintenance - Raid Alert |     New     |    983501   |   s1850-dal9  |
| CS1937302 | Required Maintenance - Raid Alert |     New     |    590929   |   s2415-dal9  |
| CS1938387 | Required Maintenance - Raid Alert |     New     |    984335   |   s1107-dal9  |
| CS1798830 | Required Maintenance - Raid Alert |     New     |    610383   |  s3180-dal09  |
| CS1904128 | Required Maintenance - Raid Alert |     New     |    888791   |   s525-dal9   |
......
......

**Expected behavior**
get_cases() should include all searched cases align with total_count in returned json

**Screenshots**

**Must gather (please complete the following information):**
python3.7
ibm-platform-service>0.17.7

**Additional context**
N/A

**Comment by padamstx** [comment]
@DanielWu0626 Hi, I think you just need to use the pagination-related parameters (`offset` and `limit`) in order to retrieve results one page at a time.
You can see these parameters described in the [on-line reference](https://cloud.ibm.com/apidocs/support-center/case-management?code=python#getcases).
This would be similar to reading the contents of a file one block at a time until you reach end of file.
You would need a loop that calls the get_cases operation and accumulates the returned results in a list.  The first time through the loop specify offset=0 and limit=<your page size> (default=100 as you have seen).   Save off the returned results in a list, then call get_cases() again, but this time increment your offset by the # of items returned in the first call.  Then repeat that until you have received the total number of results (the response's total_count property provides that value).

Hope this helps

**Comment by DanielWu0626** [comment]
thank you @padamstx , let me try again !

**Comment by padamstx** [comment]
@DanielWu0626 your question via slack:
```
I still don’t quite understand offset and limit use, in my code the first time calling get_cases() I did include offset=0 and limit=300,  the real records it should include in returned json is only 100. I pay concern on this “limit”, does it mean we cannot fill the cases in one json more than 100 records ?
```
Based on the behavior that you're seeing, I would guess that the maximum page-size supported by that operation is 100, but I'm trying to confirm that with the service team.  Perhaps just try limit=100.

here is a code snippet from the SDK project's integration test that demonstrates how to do pagination with the get_cases() method:
```
        cases = []
        offset = 0
        page_size = 10
        more_results = True
        while more_results:
            response = self.service.get_cases(
                offset=offset,
                limit=page_size,
            )
            case_list = response.get_result()
            if len(case_list.get('cases')) > 0:
                cases.extend(case_list.get('cases'))
                offset += len(case_list.get('cases'))
            else:
                more_results = False

        print('get_cases returned a total of {} cases'.format(len(cases)))
        print('returned cases\n{}'.format(json.dumps(cases, indent=2)))
```

**Comment by padamstx** [comment]
Closing this issue as the pagination example above should be sufficient to retrieve multiple pages of results.
